src/sources/generator.rs

```diff
@@ -0,0 +1,195 @@
+use crate::{
+    event::Event,
+    shutdown::ShutdownSignal,
+    topology::config::{DataType, GlobalOptions, SourceConfig, SourceDescription},
+};
+use futures::{
+    compat::Future01CompatExt,
+    future::{FutureExt, TryFutureExt},
+    stream::StreamExt,
+};
+use futures01::{future::Future, stream::iter_ok, sync::mpsc, Sink};
+use serde::{Deserialize, Serialize};
+use std::time::Duration;
+use tokio::time::interval;
+
+#[derive(Clone, Debug, Default, Deserialize, Serialize)]
+#[serde(deny_unknown_fields)]
+pub(self) struct GeneratorConfig {
+    #[serde(default)]
+    sequence: bool,
+    lines: Vec<String>,
```

Comment:
One possible extension is to support reading the lines from a file, in addition to the possibility to specify it in the configuration file. This would differ from the `file` source because after the file is read to the end, it would be read from start again, indefinitely.

Reply:
That's a great idea. So I'm clear, are you suggesting to keep the file open and keep re-reading from it, or just read it once at initialization and dump those lines?

Reply:
I think the first approach is more universal, as it supports arbitrary large files without the risk of going out of memory. In order to make it fast for small files, it might be possible to implement it using `mmap`.

Reply:
I like the file idea too! But I think we should forgo adding that right now. I assume that would complicate things. Also, the name `generator` implies that it's generating logs, not reading them.